rafthttp/http.go

```diff
@@ -72,6 +74,15 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if err := checkVersionCompability(serverVersion(r.Header), minClusterVersion(r.Header)); err != nil {
+		log.Printf("rafthttp: request ignored due to %v", err)
+		http.Error(w, errMsgIncompatibleVersion, http.StatusForbidden)
```

Comment:
I think it fits into both:
403: The server understood the request, but is refusing to fulfill it. Authorization will not help and the request SHOULD NOT be repeated.
412: The precondition given in one or more of the request-header fields evaluated to false when it was tested on the server.

403 sounds more severe and demands users to check version problem. So I choose forbidden here. Any suggestion?